pkg/controller/autoscaling/elasticsearch/policy.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package elasticsearch
+
+import (
+	"context"
+
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/tracing"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+	"github.com/go-logr/logr"
+	"go.elastic.co/apm"
+)
+
+// updatePolicies updates the autoscaling policies in the Elasticsearch cluster.
+func updatePolicies(
+	ctx context.Context,
+	log logr.Logger,
+	autoscalingSpec esv1.AutoscalingSpec,
+	esclient client.AutoscalingClient,
+) error {
+	span, _ := apm.StartSpan(ctx, "update_autoscaling_policies", tracing.SpanTypeApp)
+	defer span.End()
+	// Cleanup existing autoscaling policies
+	if err := esclient.DeleteAutoscalingPolicies(ctx); err != nil {
```

Comment:
This is something I didn't try to optimize yet. The API has been designed to be used this way, but I agree we should try to be clever here. Since this is an optimization I would track it with a dedicated issue and a separate PR ?